llvm/test/Transforms/InstCombine/fold-fcmp-trunc.ll

```diff
@@ -0,0 +1,11 @@
+; RUN: opt -passes=instcombine -S < %s | FileCheck %s
+
+
+;CHECK-LABEL: @src(
+;CHECK: %result = fcmp oge double %0, 1.000000e+02
+;CHECK-NEXT: ret i1 %result
+define i1 @src(double %0) {
+    %trunc = fptrunc double %0 to float
+    %result = fcmp oge float %trunc, 1.000000e+02
+    ret i1 %result
+}
```

Comment:
Test more combinations of types, including vectors. Also test flag preservation behavior

Reply:
A test of `fp128`, `x86_fp80`, or `ppc_fp128` in particular would be helpful.

Reply:
> Test more combinations of types, including vectors. Also test flag preservation behavior
@arsenm Sorry, could you give me a hint what 'flag preservation behavior' means in IR?

Reply:
I mean the flags on the compare should be preserved, and you don't have any tests using fast math flags. e.g. https://alive2.llvm.org/ce/z/uQr4-J

Reply:
> A test of `fp128`, `x86_fp80`, or `ppc_fp128` in particular would be helpful.

Sorry for late. 

I tried to test these types, but an error is generated whose message is "floating point constant does not have type 'fp128'". 

The error is emitted in the parse stage, and more modifications might be required to be conducted.

Should i ignore this problem, or if there are better solutions?

Reply:
The IR syntax for floating point constants is unnecessarily painful. You need to use a different prefix and have the correct hex length depending on the format. For this I usually just write fpext from a reasonable FP type to the long one, and run it through instsimplify to see how it should be printed